src/core/CheckBox/CheckBox.tsx

```diff
@@ -0,0 +1,28 @@
+import { Checkbox } from '@material-ui/core';
+import React, { useState } from 'react';
+// import clsx from 'clsx';
+import CheckOutlinedIcon from '@material-ui/icons/CheckOutlined';
+import { useStyles } from './styles';
+import { BaseCheckboxProps } from './base';
+
+interface CheckBoxProps extends BaseCheckboxProps {
+  checked: boolean;
+}
+
+const CheckBox: React.FC<CheckBoxProps> = ({ disabled, checked }) => {
+  const classes = useStyles();
+  const [check, setChecked] = useState<boolean>(checked);
+  return (
+    <Checkbox
+      data-testid="checkbox"
+      className={classes.root}
+      checkedIcon={<CheckOutlinedIcon />}
+      disabled={disabled}
+      checked={check}
+      icon={<span></span>}
```

Comment:
This is not required. It's an empty span.

Reply:
@S-ayanide This icon appears when checkbox is unchecked and when I delete that icon design breaks,this empty span is required

Reply:
I think there is a way around. if you take a look at the third checkbox from `Basic Checkbox` [here](https://material-ui.com/components/checkboxes/#checkbox) you'll find they don't use icons as a parameter at all still you can check and uncheck without breaking the design.

Reply:
Can we have a workaround for this `icon` parameter! Did the above reference help?